src/lib/packages.ts

```diff
@@ -92,7 +92,7 @@ export class AllPackages {
 	}
 
 	tryGetTypingsData({ name, majorVersion }: PackageId): TypingsData | undefined {
```

Comment:
We should clarify what a `PackageId` is. The interface is possibly misnamed since it only refers to `@types` packages and not any npm package. As of this PR, it looks like `name` in PackageId should always be `@foo/bar` not `foo__bar`? Might want to convert it to a class so we could assert that in the constructor.

Reply:
What should these type reference directives contain? `@foo/bar` or `foo__bar`?
https://github.com/Microsoft/types-publisher/blob/42307e76c34c7e0de3e4d8fd36d4f93c9ee37a75/src/lib/module-info.ts#L31-L33

Docs at https://github.com/DefinitelyTyped/DefinitelyTyped#im-writing-a-definition-that-depends-on-another-definition-should-i-use-reference-types--or-an-import does not mention that

Reply:
I found some inconsistencies while trying figure this out:
* some packages contain the imports like `import ... from "foo__bar"` instead of `import ... from "@foo/bar"`
  * `@types/material-components-web`
  * `@types/material__*`
  * `@types/ember`
* some packages contain types references like `<reference types="foo_bar" />` instead of `<reference types="@foo/bar" />`
  * `ember`
  * `feathersjs__primus` and `feathersjs__socketio`

All this causes `getModuleInfo()` return an inconsistent set of dependencies to begin with.
https://github.com/Microsoft/types-publisher/blob/42307e76c34c7e0de3e4d8fd36d4f93c9ee37a75/src/lib/module-info.ts#L10

Reply:
@andy-ms Should I just leave this as it is and maybe create a new issue for this so the rest can be merged as it fixes the original problem?